lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -604,21 +603,41 @@ def make_node_file(self):
                 txt += " ifhn=%s" % node
             mpdboottxt += "%s\n" % txt
 
+        mpdfn = os.path.join(self.mympirundir, 'mpdboot')
+        try:
+            open(mpdfn, 'w').write(mpdboottxt)
+            self.mpdboot_node_filename = mpdfn
+            self.log.debug("make_node_file: wrote mpdbootfile %s:\n%s", mpdfn, mpdboottxt)
+        except Exception as err:
+            msg = 'make_node_file: failed to write mpbboot file %s: %s' % (mpdfn, err)
+            self.log.raiseException(msg)
+
+
+    def make_machine_file(self):
+        """
+        Make the machinefile.
+
+        Parses the list of nodes that run an MPI process and writes this information to a machinefile.
+        """
+        if not self.mympirundir:
+            self.make_mympirundir()
+
+        if self.mpinodes is None:
+            self.set_mpinodes()
+
         nodetxt = '\n'.join(self.mpinodes)
         nodefn = os.path.join(self.mympirundir, 'nodes')
-        mpdfn = os.path.join(self.mympirundir, 'mpdboot')
+
         try:
             open(nodefn, 'w').write(nodetxt)
```

Comment:
same here, properly close the file handle, use `IOError`, move stuff not related to writing the file outside of the `try`/`except`